### PR TITLE
adding an abuse control module, for now only checking for absolute number of keeps

### PR DESCRIPTION
**Pull request by eishay**
more to come /cc @ymatsuda @andrewconner

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsAbuseController.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    absoluteAlert: Int,
+    absoluteError: Int,
+    bookmarkRepo: BookmarkRepo,
+    db: Database) {
+
+  if (absoluteAlert >= absoluteError) throw new IllegalStateException(s"absolute alert $absoluteAlert is larger then error $absoluteError")
+
+  implicit val dbMasterSlave = Database.Slave
+
+  def inspact(userId: Id[User], newKeepCount: Int): Unit = {
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsAbuseController.scala:21]: `inspact` => `inspect`

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsAbuseController.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import com.keepit.common.db.Id
+import com.keepit.model.User
+import com.google.inject.Inject
+import com.keepit.model.BookmarkRepo
+import com.keepit.common.db.slick.Database
+
+class AbuseControlException(message: String) extends Exception(message)
+
+class KeepsAbuseController @Inject() (
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsAbuseController.scala:11]: `KeepsAbuseController` => `KeepsAbuseCommander` (it's in the Commander package)

Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsAbuseController.scala:11]: the class stands for "abuse control" but I do get the point. i'll rename

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsAbuseController.scala:11]: Ahh. How about an `AbuseCommander`?

**Pull request closed without merging**